lib/manageiq_performance/middleware.rb

```diff
@@ -90,15 +90,30 @@ def save_report env, prefix, long_form, short_form
     end
   end
 
-  def self.profile name = nil, &code
-    name ||= caller.first.match(/`(|.*\s)([a-z_\(\)]+)>?'$/)[2].gsub(/[^a-z_]/,'')
+  def self.profile *args, &code
+    options = args.delete_at(args.find_index { |arg| arg.is_a? Hash } || args.size + 1) || {}
```

Comment:
Yeah, this was dumb... not sure where my head was at.  Will fix when I merge #28 with some of the changes you suggested.